src/main/java/com/google/sps/data/PlayerDatabase.java

```diff
@@ -84,21 +92,16 @@ public static Entity createEntityFromPlayer(Player player) {
 
   // get entity
   public Entity getCurrentPlayerEntity() throws LoggedOutException {
-    if (!isLoggedIn) {
+    if (!this.isLoggedIn) {
       throw new LoggedOutException();
     }
-    String email = userEmail;
-    Query query =
-        new Query(ENTITY_QUERY_STRING)
-            .setFilter(
-                new Query.FilterPredicate(ID_QUERY_STRING, Query.FilterOperator.EQUAL, userID));
-    PreparedQuery results = datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      if (email.equals(entity.getProperty(EMAIL_QUERY_STRING).toString())) {
-        return entity;
-      }
+    Key key = KeyFactory.createKey(ENTITY_QUERY_STRING, this.userID);
+    try {
+      Entity entity = this.datastore.get(key);
+      return entity;
+    } catch (EntityNotFoundException e) {
+      throw new LoggedOutException();
```

Comment:
I am not understanding this exception here. Wouldn't this be caught by the `if` at the beginning of the method?

Reply:
That's true, probably unnecessary.